subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/OperationDependenciesProvider.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling.internal.provider.runner;
+
+import org.gradle.api.Task;
+import org.gradle.api.execution.TaskExecutionGraph;
+import org.gradle.api.execution.TaskExecutionGraphListener;
+import org.gradle.api.internal.project.ProjectInternal;
+import org.gradle.execution.plan.ExecutionDependencies;
+import org.gradle.execution.plan.TransformationNodeIdentifier;
+import org.gradle.execution.taskgraph.TaskExecutionGraphInternal;
+import org.gradle.tooling.internal.protocol.events.InternalOperationDescriptor;
+import org.gradle.util.Path;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import static java.util.stream.Collectors.toCollection;
+
+class OperationDependenciesProvider implements TaskExecutionGraphListener {
```

Comment:
I think this is more of a `Resolver` than a `Provider`.

Reply:
I don't think you can, not anymore. :) But now you're overloading `Provider`.

I think we should just that since Gradle is built on lazy evaluation we will need the concept of resolution in many places. I actually think that using the same term for the same process is a good thing.

Reply:
Done in 2618409e25a.